docs/guides/advanced-routing.md

```diff
@@ -0,0 +1,332 @@
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/nginxinc/nginx-kubernetes-gateway/main/examples/advanced-routing/coffee.yaml
+```
+
+### Deploy the Gateway API resources for the coffee applications
+
+The Gateway resource is typically deployed by the [cluster operator][roles-and-personas]. To deploy the Gateway:
+
+```shell
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: cafe
+spec:
+  gatewayClassName: nginx
+  listeners:
+  - name: http
+    port: 80
+    protocol: HTTP
+EOF
+```
+
+This Gateway defines a single listener on port 80. Since no hostname is specified, this listener matches on all hostnames.
+
+The HTTPRoute is typically deployed by the [application developer][roles-and-personas]. To deploy the `coffee` HTTPRoute:
+
+```shell
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: coffee
+spec:
+  parentRefs:
+  - name: cafe
+    sectionName: http
+  hostnames:
+  - cafe.example.com
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /coffee
+    backendRefs:
+    - name: coffee-v1-svc
+      port: 80
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /coffee
+      headers:
+      - name: version
+        value: v2
+    - path:
+        type: PathPrefix
+        value: /coffee
+      queryParams:
+      - name: TEST
+        value: v2
+    backendRefs:
+    - name: coffee-v2-svc
+      port: 80
+EOF
+```
+
+This HTTPRoute has a few important properties:
+
+- The `parentRefs` references the Gateway resource that we created, and specifically defines the `http` listener
+to bind to, via the `sectionName` field.
+- `cafe.example.com` is the hostname that is matched for all requests to the backends defined in this HTTPRoute.
+- The first rule defines that all requests with the path prefix `/coffee` and no other matching conditions are sent
+to the `coffee-v1` Service.
+- The second rule defines two matching conditions. If *either* of these conditions match, requests are forwarded to
+the `coffee-v2` Service:
+  - Request with the path prefix `/coffee` and header `version=v2`
+  - Request with the path prefix `/coffee` and the query parameter `TEST=v2`
+    If you want both conditions to be required, you can define headers and queryParams in the same match object.
+
+[roles-and-personas]: https://gateway-api.sigs.k8s.io/concepts/roles-and-personas/#roles-and-personas_1
+
+### Send traffic to coffee
+
+Using the external IP address and port for NGINX Kubernetes Gateway, we can send traffic to our coffee
+applications.
+
+> Note: If you have a DNS record allocated for `cafe.example.com`, you can send the request directly to that
+> hostname, without needing to resolve.
+
+```shell
+curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffee
+```
+
+This request should receive a response from the server `coffee-v1`.
```

Comment:
perhaps the Pod instead of server would suit better here? (and in similar places)